utils/misc.ml

```diff
@@ -17,6 +17,13 @@ exception Fatal_error
 let fatal_error msg =
   prerr_string ">> Fatal error: "; prerr_endline msg; raise Fatal_error
 
+let fatal_errorf fmt =
+  let (_ : string) = Format.flush_str_formatter () in
+  Format.kfprintf (fun ppf ->
+      fatal_error (Format.flush_str_formatter ()))
+    Format.str_formatter
```

Comment:
And I agree with @gasche, str_formatter should be avoided. Next thing you know, someone uses the compiler concurrently (like, in jsoo) and he gets mangled error messages.